packages/peregrine/lib/Router/resolveUnknownRoute.js

```diff
@@ -99,6 +99,21 @@ function remotelyResolveRoute(opts) {
  * @returns {Promise<{type: "PRODUCT" | "CATEGORY" | "CMS_PAGE"}>}
  */
 function fetchRoute(opts) {
+    const storeCodes = AVAILABLE_STORE_VIEWS.map(store => store.code);
```

Comment:
This needs to always match the result from the `availableStores` query used to generate the store switcher options otherwise it'll be possible that we miss a new route + new store.

I'm betting you can repro this if you:

1) Build/deploy/visit a page.
2) Add a new store code to you backend that will result in a new entry of `availableStores`
3) Clear cache but don't "rebuild" your pwa studio bundle.
4) Refresh the browser, use the store switcher to change to the new store.
5) I'm betting that it will try to resolve that full route here since it won't get trimmed below.

Reply:
Discussed with @davemacaulay. This is a trade off where we pre-optimize by generating this global at build time using GQL. 

The upside is that we can render routes without having to first make an `availableStores` query.

The downside is that implementers must re-build/deploy their PWA when they create a new store and want it accessible in the UI.